### PR TITLE
feat(ui5-flexible-column-layout): enable resize via separators drag-and-drop

### PR DESCRIPTION
**Pull request by kineticjs**
Allow the user то customize the column width by drag-and-drop of the column separators.

As the user drags to resize, the following is automatically ensured: 
-- If the new column width correspond to a new FCLLayout, update the "layout" property accordingly. Ensure the current width distribution always corresponds to a valid layout (by additionally re-adjusting widths where necessary, to correctly match new FCL layout). 
-- Ensure the column width is never below the minimum 312 px widths

**BREAKING CHANGE**:

1. The `arrowUsed` and `arrowsUsed` parameters of the `layoutChange` event are now removed and replaced with `separatorsUsed`

Before:

```js
fcl.addEventListener("ui5-layout-change", function(e) { const isUserInteraction = e.detail.arrowsUsed; ... }
```

Now:

```js
fcl.addEventListener("ui5-layout-change", function(e) { const isUserInteraction = e.detail.separatorsUsed; ... }
```

2. The `resize` parameter of the `layoutChange` event is renamed to `resized`

Before:

```js
fcl.addEventListener("ui5-layout-change", function(e) { const isGlobalResize = e.detail.resize; ... }
```

Now:

```js
fcl.addEventListener("ui5-layout-change", function(e) { const isGlobalResize = e.detail.resized; ... }
```

3. The following fields of the `FCLAccessibilityAttributes` type are removed:

* startArrowContainer
* endArrowContainer
* startArrowLeft
* startArrowRight
* endArrowLeft
* endArrowRight
  and replaced with the following new fields:
* startSeparator
* endSeparator

Before:

```js
fcl.accessibilityAttributes = {
    startArrowLeft:  {
      name: "Collapse products list",
    },
    startArrowRight: {
      name: "Expand products list",
    },
    endArrowLeft: {
      name: "Expand product detailed information",
    },
    endArrowRight:  {
      name: "Collapse product detailed information",
    },
    startArrowContainer: {
      role: "navigation",
      name: "Start Arrow Container",
    },
    endArrowContainer: {
      role: "navigation",
      name: "End Arrow Container",
    },
};
```

Now:

```js
fcl.accessibilityAttributes = {
    startSeparator: {
      role: "navigation",
      name: "Start Separator",
    },
    endSeparator: {
      role: "navigation",
      name: "End Separator",
    },
};
```

4. The public `hideArrows` property is renamed to `disableResizing`

Before:

```
<ui5-flexible-column-layout hide-arrows/>
```

Now:

```
<ui5-flexible-column-layout disable-resizing/>
```

### CODE REVIEW

File: packages/fiori/src/FlexibleColumnLayout.ts [hunk 1]
```diff
@@ -460,39 +472,436 @@ class FlexibleColumnLayout extends UI5Element {
...
+		let selectedColumnToResize,
+			columnWidthDelta;
+
+		if (isRTL) {
+			offsetX = -offsetX;
+		}
+
+		if (isStartSeparator) {
+			selectedColumnToResize = COLUMN.START;
+			// move in direction start-to-end expans start column
```
Comment (plamenivanov91) [inline packages/fiori/src/FlexibleColumnLayout.ts:628]: typo -> expands

**Review by plamenivanov91 — CHANGES_REQUESTED** [review]
It works great, check out my minor comments.

Also, when testing with JAWS 2024 I noticed that separators are immovable in virtual cursor mode (I guess this is expected), but when I turn off VC and start moving the separators with the arrow buttons nothing is announced, even when the whole layout is being changed (due to far right separator movement for example). Don't know if we need to comply with this feedback or additional ACC requirement needs to be implemented (in different change may be) but I just wanted to share my findings.

**Review by plamenivanov91 — APPROVED** [review]
Looks good to me. We agreed upon implementing additional ACC requirements in a future change.

File: packages/fiori/src/FlexibleColumnLayout.ts [hunk 2]
```diff
@@ -119,11 +134,19 @@ type FCLAccessibilityAttributes = {
...
  * and 3 columns for sizes bigger than 1023px.
  *
+ * **Note:** When the component displays more than one column, the minimal width of each column is 312px. Consequently, when the user drags a column separator to resize the columns, the minimal allowed width of any resized column is 312px.
+ *
  * ### Keyboard Handling
  *
  * #### Basic Navigation
  *
- * - [Space] / [Enter] or [Return] - If focus is on the layout toggle button (arrow button), once activated, it triggers the associated action (such as expand/collapse the column).
+ * When a solumn separator is focused,  the following keyboard
```
Comment (simonarangelova) [inline packages/fiori/src/FlexibleColumnLayout.ts:143]:
```suggestion
 * When a column separator is focused,  the following keyboard
```

**Comment by simonarangelova** [comment]
> Allow the user то customize the column width by drag-and-drop of the column separators.
> 
> As the user drags to resize, the following is automatically ensured: -- If the new column width correspond to a new FCLLayout, update the "layout" property accordingly. Ensure the current width distribution always corresponds to a valid layout (by additionally re-adjusting widths where necessary, to correctly match new FCL layout). -- Ensure the column width is never below the minimum 312 px widths
> 
> **BREAKING CHANGE**:
> 
> 1. The `arrowUsed` and `arrowsUsed` parameters of the `layoutChange` event are now removed and replaced with `separatorsUsed`
> 
> Before:
> 
> ```js
> fcl.addEventListener("ui5-layout-change", function(e) { const isUserInteraction = e.detail.arrowsUsed; ... }
> ```
> 
> Now:
> 
> ```js
> fcl.addEventListener("ui5-layout-change", function(e) { const isUserInteraction = e.detail.separatorsUsed; ... }
> ```
> 
> 2. The `resize` parameter of the `layoutChange` event is renamed to `resized`
> 
> Before:
> 
> ```js
> fcl.addEventListener("ui5-layout-change", function(e) { const isGlobalResize = e.detail.resize; ... }
> ```
> 
> Now:
> 
> ```js
> fcl.addEventListener("ui5-layout-change", function(e) { const isGlobalResize = e.detail.resized; ... }
> ```
> 
> 3. The following fields of the `FCLAccessibilityAttributes` type are removed:
> 
> * startArrowContainer
> * endArrowContainer
> * startArrowLeft
> * startArrowRight
> * endArrowLeft
> * endArrowRight
>   and replaced with the following new fields:
> * startSeparator
> * endSeparator
> 
> Before:
> 
> ```js
> fcl.accessibilityAttributes = {
>     startArrowLeft:  {
>       name: "Collapse products list",
>     },
>     startArrowRight: {
>       name: "Expand products list",
>     },
>     endArrowLeft: {
>       name: "Expand product detailed information",
>     },
>     endArrowRight:  {
>       name: "Collapse product detailed information",
>     },
>     startArrowContainer: {
>       role: "navigation",
>       name: "Start Arrow Container",
>     },
>     endArrowContainer: {
>       role: "navigation",
>       name: "End Arrow Container",
>     },
> };
> ```
> 
> Now:
> 
> ```js
> fcl.accessibilityAttributes = {
>     startSeparator: {
>       role: "navigation",
>       name: "Start Separator",
>     },
>     endSeparator: {
>       role: "navigation",
>       name: "End Separator",
>     },
> };
> ```
> 
> 4. The public `hideArrows` property is renamed to `disableResizing`
> 
> Before:
> 
> ```
> <ui5-flexible-column-layout hide-arrows/>
> ```
> 
> Now:
> 
> ```
> <ui5-flexible-column-layout disable-resizing/>
> ```

**Comment by kineticjs** [comment]
Thanks, done

**Pull request merged**